src/renderer/components/inputs/DropDownInput.tsx

```diff
@@ -1,55 +1,41 @@
-import { Select } from '@chakra-ui/react';
-import { ChangeEvent, memo, useEffect } from 'react';
+import { Box } from '@chakra-ui/react';
+import { memo, useCallback } from 'react';
+import { Checkbox } from './elements/Checkbox';
+import { DropDown } from './elements/Dropdown';
+import { WithLabel } from './InputContainer';
 import { InputProps } from './props';
 
-type DropDownInputProps = Pick<
-    InputProps<'dropdown', string | number>,
-    'value' | 'setValue' | 'input' | 'isLocked'
->;
+type DropDownInputProps = InputProps<'dropdown', string | number>;
 
 export const DropDownInput = memo(({ value, setValue, input, isLocked }: DropDownInputProps) => {
-    const { options, def } = input;
+    const { options, def, label } = input;
 
-    const handleChange = (event: ChangeEvent<HTMLSelectElement>) => {
-        setValue(options[Number(event.target.value)]?.value ?? def);
-    };
+    const reset = useCallback(() => setValue(def), [setValue, def]);
 
-    // set default
-    useEffect(() => {
-        if (value === undefined) {
-            setValue(def);
-        }
-    }, [value, setValue, def]);
-
-    // reset invalid values to default
-    useEffect(() => {
-        if (value !== undefined && options.every((o) => o.value !== value)) {
-            setValue(def);
-        }
-    }, [value, setValue, options, def]);
-
-    let selection = options.findIndex((o) => o.value === value);
-    if (selection === -1) selection = 0;
+    if (options.length === 2 && options[0].option === 'Yes') {
```

Comment:
Could you make this an exhaustive check for both Yes and No, and also include checks for things like true/false?

Also, I'm thinking maybe it should actually be defined in the backend as a checkbox input that uses a boolean rather than a dropdown. Would make more sense IMO, and then you wouldn't need this check at all.

Reply:
Yes, the current yes/no detection is only a temporary solution. Thanks for pointing that out, I almost forgot to change it.

I was thinking about something like a `preferredStyle` property for dropdowns. This would allow dropdowns to communicate how they are would like to displayed. The frontend can then choose to display them differently (e.g. in the style of a checkbox).